packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -176,60 +179,101 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
 
 `nimble-table-column-text` will continue to present the same API it does today, but will derive from the base classes described above.
 
-#### Numeric column
+#### Number column
+
+Nimble will introduce `nimble-table-column-number-text` which formats a numeric field value and displays it as text. It will offer sufficient configuration to support use cases 1-4 above.
+
+##### API
```

Comment:
One SLE use case which came up at the UI WG meeting was tag columns where values are typically numbers with a known unit (e.g. bytes) and ideally they would be displayed with smartly sized units (KB, MB). But because tag values are user-provided, it's possible for units to differ within a column (some values might be volts or amps) and even for values not to be numbers.

An idea I had to support the case where units aren't consistent is to add a string attribute `suffix-field-name` (and maybe also `prefix-field-name`). It would be used instead of `suffix` (or we could document which one is rendered first if both are provided). But instead of being a constant string it would be a string field that contains the unit for the tag.

An idea I had to support the case where values are mixed types is to add a string attribute `placeholder-field-name`. It would be used instead of `placeholder` when the number value can't be rendered. Instead of being a constant string it would be a string field that contains the client-formatted string value of the tag.

Finally, we could add a new format mode `bytes` which displays byte number values with correctly scaled KB or MB.

With those 3 ideas we could support:
- the common case of byte values having nicely scaled units but maintaining numeric sorting
- non-numeric values could render as strings instead of a constant placeholder. We'd likely sort them at the end. I'm not sure if we could sort within the non-numeric values or not.
- non-byte numeric values could render their unit strings. I'm not sure how they'd be sorted; possibly interleaved with byte values. But I don't know if a sane sort order exists for values with differing units.

I'm curious for feedback on these ideas. My initial thought is to list them in the HLD but deprioritize their implementation initially.

Reply:
I don't think the proposed API will solve the use cases of the [Spec Mangement Unit Conversion feature](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/446556). They'll have numeric data uploaded in any unit (e.g. mV or V) but stored in a normalized way as "base units". When they display, users want the data rendered with its uploaded unit but (presumably) sorted by normalized unit. If data is uploaded with a mixture of units, I don't think any of my current proposals could both render those units and maintain correct sorting. I believe that doing so would require the table to include all the unit conversion logic described in that linked spec.

I think that's way out of scope so I propose we don't design for it yet.

Reply:
After taking a look at all the existing SLE column types and the Spec Management Unit Conversion HLD, my overall thoughts are:
- Your current API in the HLD seems reasonable. I'd probably start with just the current `prefix` and `suffix` properties (and not the more dynamic `-field-name` variants).
- I'm somewhat undecided if `bytes` should be a new format mode. Leaning towards it being a separate column type (especially if it doesn't need to use any of the numeric formatting options, the current `FormatBytesSizePipe` in SLE doesn't really do any special numeric formatting)

I agree with not trying to design in APIs for the unit conversion feature yet. At some point we might want a column type that handles unit suffixes and SI unit conversions but there's enough unknowns still that I don't think we should try to do that yet (like the 'base unit vs uploaded unit' mismatch that you mentioned).  
If their web service APIs will eventually return measurements as both base units and uploaded units, it probably wouldn't be too difficult to make a custom column type to handle that use case (I think base units would be `operandDataRecordFieldName` and the uploaded units would be an additionally specified field name used for display). But again, no need to design for that yet.